### DISCUSSION TITLE
How to handle many to one power connections (DC bus bar)

### DISCUSSION THREAD

**Discussion by jnicholsDLR** [netbox-community/netbox#12902] [Q&A] [answered]
Hello. I am looking for advice on how to best handle many-to-one power connections. In my case, this is a DC bus bar, which is powered by a single OCP ORv2 Power Shelf, and supplies power for around 18 servers.

For those that are not familiar with this, it is an open standard put together by the Open Compute Project community. The specific equipment I am dealing with is designed and built based on the ORv2 (Open Rack version 2) standard, but ultimately doesn't change the main question. The power shelf takes AC input, converts it to DC, and sends it to a vertical bus bar located in the rear of the rack. There are then shelves installed which interface with this DC bus and break out connections for 2 or 3 servers that can be slotted into the shelf.

The way I understand it is that Netbox only supports one-to-one connections for power. This makes absolute sense if you're dealing with typical AC power, where one outlet *should* only provide power to one device. However, this breaks down when operating with a DC bus bar. This may all be a moot point, but I figured I would ask the community in case anybody else has run into this scenario and is trying their best to track power.

**Comment by candlerb** [comment, marked as answer]
> The way I understand it is that Netbox only supports one-to-one connections for power.

No, it also supports one-to-many. Create a PDU device (in your case "bus bar") with one Power Port (inlet) and multiple Power Outlets, where all the Power Outlets are internally linked to the same Power Port: there is a field for this on Power Outlet.  It's easier if you create the Power Port first, so that when you create the Power Outlets they can be immediately associated with it.  Also note that you can create multiple Power Outlets in one go, e.g. by giving the name `OUT[1-16]`.

This modelling should work fine for distributing a single DC feed to multiple servers.  Each server connects to a different "Power Outlet" on the same PDU/bus bar device.

**Comment by jnicholsDLR** [comment]
Thank you @candlerb, I knew there was a more obvious way to approach this that I was completely overlooking. Here I am with all these AC PDUs and I hadn't thought about making a "bus bar" PDU device. Thanks for the suggestion!
